src/libraries/System.Text.Json/tests/Serialization/ExtensionDataTests.cs

```diff
@@ -220,6 +221,12 @@ private class ClassWithTwoExtensionProperties
             public Dictionary<string, object> MyOverflow2 { get; set; }
         }
 
+        private class ClassWithNotSupportedExtensionPropertyImmutableDictionary
+        {
+            [JsonExtensionData]
+            public ImmutableDictionary<string, object> MyOverflow { get; set; }
```

Comment:
We should test what happens when we have a custom converter specified -

-  with an attribute on the dictionary type
- with an attribute on the extension property 
- added at runtime on the `options.Converters` collection